### ISSUE TITLE
Wrong flag for Ingrian

### ISSUE THREAD

**Issue by Guybrush88** [Tatoeba/tatoeba2#3129]
As reported by Thadh on the wall:

> Hi!
> 
> I've noticed the flag used for Ingrian is incorrect.
> 
> Currently the flag of Ingria is used ( as [https://commons.wikimedia.org/w...germanland.svg](https://commons.wikimedia.org/wiki/File:Flag_of_Ingermanland.svg) ), but this flag is only used by the Ingrian Finns, which speak a dialect of Finnish.
> 
> The orthodox Izhorians, which speak Ingrian, instead use this flag ( [https://commons.wikimedia.org/w...Flag_igora.svg](https://commons.wikimedia.org/wiki/File:Flag_igora.svg) ). I think it's best to change this.
> 
> Thanks in advance,
> 
> Thadh

https://tatoeba.org/it/wall/show_message/40618#!#message_40618

**Comment by DJ-Saidez** [comment]
Maybe lang-request may have been a better label, since this involves language icons. I'll get on it right now!

**Comment by DJ-Saidez** [comment]
I'm having difficulty changing the flag. The design is quite complex with hundreds of line segments, and exceeds the 4 K limit for file size, even after attempts to optimize and simplify the file.

**Comment by jiru** [comment]
@DJ-Saidez I will have a look at it.

**Comment by jiru** [comment]
The [Wikimedia page of the flag](https://commons.wikimedia.org/wiki/File:Flag_igora.svg) says:

> This [SVG (Scalable Vector Graphic)](https://commons.wikimedia.org/wiki/Help:SVG) appears to have been inadequately vectorized, for example, by [auto-tracing](https://en.wikipedia.org/wiki/en:Vectorization_(image_tracing)), and may require [revectorization](https://commons.wikimedia.org/wiki/Commons:Media_for_cleanup#PNG_or_JPEG_images_that_should_be_SVG) to meet quality standards.

So somebody will have to take on the work of "revectorization", or find another file, before we can use it.

**Comment by jiru** [comment]
Another way is to "solve" this problem is to export that SVG into a 20×30px PNG image, and embed it into an SVG, just like many of our existing flags are, because they used to be 20×30px PNG images that were later "converted" to SVG. I am afraid a scaled down version of that flag will look terrible though.

**Comment by DJ-Saidez** [comment]
Yes, I thought of that and tried it, and it was basically unrecognizable, I thought it was the flag of Moldova.
I'll let Thadh know, thanks for giving it a look!

**Comment by jiru** [comment]
Thank you for helping with the flag, @franklindyer 😃

> where does the 4KB size limit on the flags come from?

The 4KB is rather arbitrary, and it is just there to make sure we avoid adding oversized flags. We have this limit because it’s just too easy to download any image from Wikimedia (or anywhere else) and add it to the repository without paying attention to whether it is properly compressed and optimized. Since Tatoeba serves hundreds of flags, if we don’t pay attention to the size of each flag, it can easily become bloated.

If any flag exceeds the limit, the automated build (CI) fails. The CI is defined in `./Jenkinsfile`, which starts by running `./check.sh`, which checks for the flag sizes. The CI is run on every push to a PR branch or to the main branch. It takes a few minutes to run and then GitHub displays a green check or a red cross depending on the result.

The limit used to be 1KB, then it was later raised to 2KB, 3KB and now it is 4KB. We could raise it again, but let’s see.

> I don't see much hope for a faithful representation of the Izhorian flag in an SVG under 4.0KB

There are many techniques to further reduce the file size of SVGs.

1. The most straightforward is to use a markup optimizer such as https://svgomg.net/. (Beware that, by default, svgomg displays the compressed filesize, not the actual filesize.)
2. You can put repeating patterns inside a `<g id="xxx">` markup and declare translated/mirrored/rotated copies of them with `<use href="#xxx" transform="..." />`. Just like refactoring code, you can refactor markup. See `hin.svg`, `cmn.svg` or `yue.svg` for an example.
3. When the width of a polygon is constant, you can use very think strokes instead of polygons. See `heb.svg` for an example.
4. You can adjust the coordinates reference `viewBox="..."` in a way that makes most numbers round and small. See `kpv.svg` or `que.svg` for an example.
6. If the flag includes a very intricate drawing (such as an emblem) on an otherwise flat background, you can insert the intricate drawing as a base64-encoded raster image `<image href="data:..." />`. See `nah.svg` or `spa.svg` for an example.
7. You can decide to remove/simplify some parts of the original flag, if you believe it’s just too much detail. See `por.svg` and `por.svgsrc` for an example.

I am not sure if I need to say "beware the optimization rabbit hole" or "you don’t need to do *all* that". 😆 But I have the intuition the Ingrian flag could benefit a lot from these techniques, potentially lowering its size under 4KB while staying faithful to the original. It would be ideal to find the original flag construction sheet though.

**Comment by franklindyer** [comment]
Thanks for the pointers! I have learned a lot about SVGs from tinkering with this ;-)

I made a new hand-traced version that is _quite_ faithful to the original and sits at about 3.5 KB. Many of the techniques you listed were very useful. Besides those, I noticed that the editor I was using included a lot of extraneous `style` attributes in exported SVGs. And even for some of the necessary style attributes, I was able to eliminate repetition by using the fact that `style` attributes on `g` elements are inherited by their child nodes.

Here's the new SVG:

![Image](https://github.com/user-attachments/assets/0c717871-c5c5-44ed-bf45-4131da9dcee2)

I'll go ahead and open a PR for this change.
